src/main/java/com/google/sps/data/GivrUser.java

```diff
@@ -104,7 +104,14 @@ public static void updateUserInDatastore(String identifyingProperty, String iden
     datastore.put(entity);
   }
 
-  public static GivrUser getUserByIdOrEmail(String userId, String userEmail) {
+  private static GivrUser getUserByIdOrEmail(String userId, String userEmail) {
+    if (userId == null) {
+      userId = "";
+    } 
+    if (userEmail == null) {
+      userEmail = "";
```

Comment:
won't this match any userEmails that are empty?

Perhaps start entityRetrievedWith* as null and then set them if the corresponding input param is valid?

Reply:
This makes sense. I made the appropriate changes to handle null userID (and userEmail).